src/include/extern.h

```diff
@@ -2544,6 +2544,7 @@ static WT_INLINE void __wt_scr_free(WT_SESSION_IMPL *session, WT_ITEM **bufp);
 static WT_INLINE void __wt_seconds(WT_SESSION_IMPL *session, uint64_t *secondsp);
 static WT_INLINE void __wt_seconds32(WT_SESSION_IMPL *session, uint32_t *secondsp);
 static WT_INLINE void __wt_set_shared_double(double *to_set, double value);
+static WT_INLINE void __wt_set_shared_maximum64(uint64_t *shared_value, uint64_t new_val);
```

Comment:
The combination of these two names is unfortunate. If read consistently, it implies that either `maximum64` is a type (clearly not), or that `double` is an operation (which is plausible, atomic `2 * v` or `v << 1` is a reasonable operation). So maybe this should be named something like `__wt_set_max_shared_u64` would be better? That would keep the pattern `__wt_OP_shared_TYPE`. Also by putting `u` in there it gives a clean way to extend to support signed types in the future.

Also I suggest returning the old value of `shared_value` prior to the write. I think a lot of algos will need to know at least if they increased it or were a no-op.

Reply:
[20c04cb](https://github.com/wiredtiger/wiredtiger/pull/10740/commits/20c04cb0c22d6ced07dc4804d9c1d2d97fcb6310) I haven't yet implemented the `shared_value` return, I think sometimes things are better added as needed rather than proactively? I need to fish around the codebase for other places we set shared maximums.